alchemist/alchemist-grid/src/main/java/it/unibo/alchemist/grid/simulation/RemoteResultImpl.java

```diff
@@ -0,0 +1,94 @@
+package it.unibo.alchemist.grid.simulation;
+
+import java.io.FileNotFoundException;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Charsets;
+
+import it.unibo.alchemist.grid.config.SimulationConfig;
+import it.unibo.alchemist.grid.exceptions.RemoteSimulationException;
+
+/**
+ * {@link RemoteResult} implementation.
+ *
+ */
+public class RemoteResultImpl implements RemoteResult {
+
+    private final String result;
+    private final UUID workerNode;
+    private final Optional<Throwable> simulationErrors;
+    private final SimulationConfig config;
+
+    /**
+     * 
+     * @param result Result file's content as string
+     * @param workerNode UUID of worker node that has done the simulation
+     * @param simulationErrors Simulation's errors
+     * @param config Simulation's specific config
+     */
+    public RemoteResultImpl(final String result, final UUID workerNode, final Optional<Throwable> simulationErrors,
+            final SimulationConfig config) {
+        this.result = result;
```

Comment:
Check non-null parameters with `Objects.requireNonNull`. This would also allow a simplified `equals`.